src/api/spec/controllers/webui/request_controller_spec.rb

```diff
@@ -274,33 +279,318 @@
         expect(bs_request.reload.state).to eq(:revoked)
       end
 
-      it 'adds submitter as maintainer' do
-        login(receiver)
-        post :changerequest, params: {
-          number: bs_request.number, accepted: 'accepted',
-          add_submitter_as_maintainer_0: "#{target_project}_#_#{target_package}" # rubocop:disable Naming/VariableNumber
-        }
-        expect(bs_request.reload.state).to eq(:accepted)
-        expect(target_package.relationships.map(&:user_id)).to include(submitter.id)
-      end
+      context 'when using the beta request show page' do
+        context 'and sending a submit request from a source package to an existing target package' do
+          it 'accepts the request and adds the submitter a maintainer when clicking the right dropdown button' do
+            login(receiver)
+            post :changerequest, params: {
+              number: bs_request.number, accepted: 'Accept and make maintainer'
+            }
+            expect(bs_request.reload.state).to eq(:accepted)
+            expect(target_package.relationships.map(&:user_id)).to include(submitter.id)
```

Comment:
I would remove this check, instead I would test that the controller returns the proper http status value. If the request is accepted we trust the BsRequest did it right. If we need to test the BsRequest behaviour we need to do it in a model spec, as it is faster and cheaper.